### ISSUE TITLE
Invalid dependency resolution

### ISSUE THREAD

**Issue by lo1tuma** [svanderburg/node2nix#90]
I’ve noticed a problem while trying to nixify [emoj 2.0.0](https://www.npmjs.com/package/emoj).

Steps to reproduce:

*The node2nix version used is the current `master` version (c544b2dfa2c37eec1e52be1ecf024864219d87e0) on macOS*

* create a `node-packages.json` which contains `[ "emoj" ]`
* run `node2nix -8 -i node-packages.json`
* run `nix-build -A emoj`
* run `./result/bin/emoj`

The `emoj` command crashes with the following exception:

```
/nix/store/zffvygcwl2zfijki3sdxa4vxbfwwh2zi-node-emoj-2.0.0/lib/node_modules/emoj/node_modules/indent-string/index.js:13
		throw new TypeError(`Expected \`count\` to be a \`number\`, got \`${typeof count}\``);
		^

TypeError: Expected `count` to be a `number`, got `string`
    at module.exports (/nix/store/zffvygcwl2zfijki3sdxa4vxbfwwh2zi-node-emoj-2.0.0/lib/node_modules/emoj/node_modules/indent-string/index.js:13:9)
    at module.exports (/nix/store/zffvygcwl2zfijki3sdxa4vxbfwwh2zi-node-emoj-2.0.0/lib/node_modules/emoj/node_modules/redent/index.js:6:9)
    at module.exports (/nix/store/zffvygcwl2zfijki3sdxa4vxbfwwh2zi-node-emoj-2.0.0/lib/node_modules/emoj/node_modules/meow/index.js:45:13)
    at Object.<anonymous> (/nix/store/zffvygcwl2zfijki3sdxa4vxbfwwh2zi-node-emoj-2.0.0/lib/node_modules/emoj/cli.js:13:13)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
```

The problem is that `redent` requires `indent-string` in a version compatible to `^2.1.0` but `node2nix` somehow decided to use version `3.2.0`.

The `package.json` in `/nix/store/zffvygcwl2zfijki3sdxa4vxbfwwh2zi-node-emoj-2.0.0/lib/node_modules/emoj/node_modules/redent/package.json` also says that `redent@1.0.0` has a dependency to `indent-string@3.2.0` but when I manually download `redent@1.0.0` (via `wget https://registry.npmjs.org/redent/-/redent-1.0.0.tgz`) and look in the `package.json` of this tarball then it requires `indent-string@^2.1.0`.

So somehow `node2nix` ignores those requirements of `redent@1.0.0`.

**Comment by svanderburg** [comment]
I have investigated the issue and it seems that you're right. It took me a while to pinpoint the cause, but I implemented a fix in revision 50d236aff15c53e0f46423c039a19b025e7fa248

You can try it by installing the development version.

I still need to do some additional tests.

**Comment by lo1tuma** [comment]
I can config, with latest `master` I can nixify and build `emoji` successfully.

Thanks for fixing this issue.

**Comment by svanderburg** [comment]
Closing this, since I released a new version: 1.6.0 that contains this bugfix.

**Comment by lo1tuma** [comment]
Awesome, thank you 🍻.
